File: src/main/java/com/partymakers/shareparty/data/persistence/friends/impl/FriendsRepositoryImpl.java
package com.partymakers.shareparty.data.persistence.friends.impl;

import com.partymakers.shareparty.data.persistence.friends.FriendPersistenceRepository;
import com.partymakers.shareparty.data.persistence.friends.entity.FriendEntity;
import com.partymakers.shareparty.domain.friends.entity.Friend;
import com.partymakers.shareparty.domain.friends.port.FriendsRepository;

import java.util.Optional;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public class FriendsRepositoryImpl implements FriendsRepository {

    private final FriendPersistenceRepository repository;

    @Override
    public Friend save(Friend friend) {
        return repository.save(FriendEntity.toPersistence(friend)).toDomain();
    }

    @Override
    public Optional<Friend> findOneById(String nickName) {
        return repository.findById(nickName).flatMap(friendEntity -> Optional.of(friendEntity.toDomain()));
    }
}

File: src/main/kotlin/com/partymakers/shareparty/data/persistence/friends/impl/FriendsRepositoryImpl.kt
package com.partymakers.shareparty.data.persistence.friends.impl

import com.partymakers.shareparty.data.persistence.friends.FriendPersistenceRepository
import com.partymakers.shareparty.data.persistence.friends.entity.FriendEntity
import com.partymakers.shareparty.domain.friends.entity.Friend
import com.partymakers.shareparty.domain.friends.port.FriendsRepository
import org.springframework.stereotype.Repository

@Repository
class FriendsRepositoryImpl(
    private val repository: FriendPersistenceRepository
) : FriendsRepository {

    override fun save(friend: Friend): Friend =
        repository.save(FriendEntity.toPersistence(friend)).toDomain()

    override fun findOneById(nickName: String): Friend? =
        repository.findByNickName(nickName)
            ?.toDomain()
}